### ISSUE TITLE
Seamless Lockwise integration in Fenix for existing Lockwise users

### ISSUE THREAD

**Issue by lime124** [mozilla-mobile/fenix#550]
## User Story
As a Lockwise user, I want to have a seamless integrated experience when using Fenix with Lockwise, so I can use them together and get the most out of having a Firefox account. 

## Acceptance Criteria
- I can use Lockwise with Fenix.
- I have a seamless experience (minimize transitions, alerts, etc)
- I am not confused about similar Fenix or Lockwise features (e.g. double prompt) 
- I want to be able to find my logins in Lockwise if autofill doesn't work.

### Dependencies
- [ ] Lockwise login improvement work is finished

Design Work
- [ ] Request for Comments from UX team
- [ ] N/A Usability testing if a new pattern or potentially confusing UI
- [x] Request for Comments from Fenix team and/or A/C, GV, Sync, etc as needed
- [x] Interaction details included
- [ ] N/A Accessibility - is this a review by someone, just keeping it in mind?
- [x] For existing UI changes, details are highlighted
- [x] Strings/content reviewed and approved
- [ ] Are they too long for the component in other locales

Deliverable
- [x] Assets uploaded to Google Drive
- [x] Links to deliverables (spec, assets) posted in comment 0

Deliverable Links

[UX Spec](https://share.goabstract.com/dc2a86f3-470a-4808-b5d7-06521148dd77?mode=design&sha=7d5036c02bd619e8b9591c3e07a3a690f8c3881f)
[Assets](https://drive.google.com/open?id=1stei4xIiQS3yFL_CQ5sXezGPv-4ISStG&authuser=amlee@mozilla.com&usp=drive_fs)

**Comment by sandysage** [comment]
## Acceptance criteria

* should work towards an integrated experience for the most seamless interaction for a user 
* should be able to download Lockbox from Fenix
* should be able to open Lockbox from Fenix
* should be contextual only be accessible from the existing "logins" flow (currently in Library)

**Comment by sandysage** [comment]
We may need another follow up issue for this user story: When I'm using Fenix, and get to a login form for one of my online accounts, I want to know how to use my Firefox saved logins to fill in login forms.

**Comment by changecourse** [comment]
I'm not sure what you mean in the latest comment, @sandysage. Can you please clarify if this is still pertinent based on the latest InVision flow?

**Comment by sandysage** [comment]
@changecourse When I go to twitter in my new Fenix browser, and I'm already signed into my Firefox Account, my current expectation is that the credentials I save in Firefox would automatically populate the username and password fields on the website. I maybe don't yet know I need to find Library -> logins to troubleshoot why my saved credentials aren't working. Is there anything we can/should introduce on the login form itself as a future story?

**Comment by changecourse** [comment]
ah, fair... similar to how we handle the identified UN/PW fields for autofill with "Search Firefox Lockbox" within it, but instead of seeing that, some form of promo for "use firefox saved login" or equivalent to launch the promo overlay? 

<img width="369" alt="screen shot 2019-03-04 at 10 14 20 am" src="https://user-images.githubusercontent.com/6164801/53750181-53801500-3e66-11e9-986c-af5ae348f43f.png">

@sandysage yes, let's add as a separate issue to handle.

**Comment by lime124** [comment]
Amy and I met with Nick and Ryan last week to go over proposed flows/UI. Amy will work continue to iterate on the final UI designs in preparation for dev.

**Comment by bbinto** [comment]
@lime124 - I don't think this is mandatory for MVP tbh - I feel Amy could/should pick up other UX issues in the meantime. See my comment here: https://github.com/mozilla-mobile/fenix/issues/87#issuecomment-472390978

**Comment by lime124** [comment]
Moving this out of the MVP work for UX.

**Comment by lime124** [comment]
@vesta0 - amy mentioned this was a desired phase two after the login manager work she's during now. do we want to put this in to the ux backlog to tackle in the near future?

**Comment by vesta0** [comment]
I raised it to Amy to make sure that she is aware of it (to ensure design alignment) but it's more of a `could` for Q4.

**Comment by lime124** [comment]
perfect - thanks!

**Comment by liuche** [comment]
@sandysage @lime124 I split this into 2 issues (#5498) so we can better address these two situations, because they can be done in parallel.

**Comment by vesta0** [comment]
Updated the description.

**Comment by ekager** [comment]
Hi @jhugman @eliserichards if we want to deep link directly to a login to edit in Lockwise by some sort of ID is that something that would be possible in Lockwise? Is there a shared ID because we are both pulling from sync? 
@AmyYLee was asking as part of the design process.

**Comment by mnoorenberghe** [comment]
> Is there a shared ID because we are both pulling from sync?

Each login has a GUID which is persistent. It's [`ServerPassword.id`](https://github.com/mozilla/application-services/blob/ba6690a5cad308fafdb4053545cdb142eb86e1b9/components/logins/android/src/main/java/mozilla/appservices/logins/ServerPassword.kt#L22) in the Rust login library.
